dashboard/app/controllers/lessons_controller.rb

```diff
@@ -35,9 +35,13 @@ def edit
   # PATCH/PUT /lessons/1
   def update
     if params[:originalLessonData]
-      current_lesson_data = JSON.generate(@lesson.summarize_for_lesson_edit)
-      old_lesson_data = params[:originalLessonData]
-      if old_lesson_data != current_lesson_data
+      current_lesson_data = @lesson.summarize_for_lesson_edit
+      old_lesson_data = JSON.parse(params[:originalLessonData])
+      current_lesson_data[:vocabularies] = current_lesson_data[:vocabularies].map {|v| v[:id]} if current_lesson_data[:vocabularies]
+      old_lesson_data['vocabularies'] = old_lesson_data['vocabularies'].map {|v| v['id']} if old_lesson_data['vocabularies']
+      current_lesson_data[:resources] = current_lesson_data[:resources].map {|v| v[:id]} if current_lesson_data[:resources]
+      old_lesson_data['resources'] = old_lesson_data['resources'].map {|v| v['id']} if old_lesson_data['resources']
```

Comment:
totally optional, but if you want to clean these lines up a bit they can each just be:
```
old_lesson_data['resources']&.map! {|v| v['id']}
```